app/controllers/leave_applications_controller.rb

```diff
@@ -23,7 +23,7 @@ def update
   end
 
   def cancel
-    if current_object.canceled?
```

Comment:
Use `#may_cancel?` provided by `aasm` to judge whether status can be transformed.